### ISSUE TITLE
Grep search results in preview window are ignored and can't be opened if line >=77 symbols

### ISSUE THREAD

**Issue by LoydAndrew** [liuchengxu/vim-clap#701]
**Environment (please complete the following information):**
 - OS: Xubuntu 20.10 x86_64
 - (Neo)Vim version: 0.5.0+ubuntu2+git202106012312-3fe7a0470-d569569c9-9efd42511~ubuntu20.10.1)
 - vim-clap version: 41d758715f1310d546d02a07ac2aa1ded2cf617e
 - Have you reproduced with a minimal vimrc: yes
 - Have you updated to the latest plugin version: yes
 - Have you upgraded to/compiled the latest Rust binary: yes

**Describe the bug**
Clap grep search results in preview window ignored if searched word is placed on line with more than 77 symbols (there is no preview in preview window and if you hit enter on such result it will create an empty buffer with a name of a file which contains this searched word).

**Clap debug**

has cargo: 1
has maple: /home/olelen/.vim/plugged/vim-clap/target/release/maple
maple info: version 0.1.25 (git v0.25-15-gff32ca1), built for x86_64-unknown-linux-gnu by rustc 1.48.0 (7eac88abb 2020-11-16).
has +python3: 1
has py dynamic module: 1
has ctags: without JSON output support
Current sync impl: Lua
 Current FileType: text
Third Party Providers: []
   Global Options:
let g:clap#autoload_dir = '/home/olelen/.vim/plugged/vim-clap/autoload'
let g:clap#provider_alias = {'gfiles': 'git_files', 'hist:': 'command_history', 'hist/': 'search_history'}
let g:clap_background_shadow_blend = 50
let g:clap_disable_bottom_top = 0
let g:clap_disable_matches_indicator = v:false
let g:clap_disable_optional_async = v:false
let g:clap_disable_run_rooter = v:false
let g:clap_enable_background_shadow = 0
let g:clap_enable_debug = 1
let g:clap_enable_icon = 1
let g:clap_forerunner_status_sign = {'done': '•', 'running': '!', 'using_cache': '*'}
let g:clap_insert_mode_only = 1
let g:clap_layout = {'relative': 'editor'}
let g:clap_multi_selection_warning_silent = 0
let g:clap_no_matches_msg = 'NO MATCHES FOUND'
let g:clap_open_action = {'ctrl-v': 'vsplit', 'ctrl-x': 'split', 'ctrl-t': 'tab split'}
let g:clap_open_preview = 'always'
let g:clap_popup_border = 'single'
let g:clap_preview_direction = 'LR'
let g:clap_preview_size = 10
let g:clap_providers_relaunch_code = '@@'
let g:clap_search_box_border_style = 'curve'
let g:clap_search_box_border_symbols = {'nil': ['', ''], 'curve': ['', ''], 'arrow': ['', '']}
let g:clap_theme = {'preview': {'ctermbg': '239', 'ctermfg': '123', 'guibg': '#2d3135', 'guifg': '#f8f8f2'}, 'display': {'ctermbg': '239', 'ctermfg': '123', 'guibg': '#2d3135', 'guifg': '#f8f8f2'}}
Provider Variables:
let g:clap_provider_dotfiles = {'source': ['~/.config/nvim/init.vim', '~/vim_keys.md', '~/.zshrc', '~/.tmux.conf', '~/.config/nvim/lua/evilline.lua', '~/.config/nvim/lua/evilline.lua'], 'sink': 'e'}
let g:clap_provider_generated_tags = {'id': 'project_tags', 'source': function('<lambda>3'), 'sink': function('<lambda>4')}
let g:clap_provider_grep_enable_icon = 1
let g:clap_provider_grep_opts = '--vimgrep --hidden -g "!node_modules/" -g "!idea" -g "!.vscode" -g "!venv" -g "!.mypy_cache" -g "!build" -g "!dist"  -g "!__pycache__" -g "!tags" -g "!*.git" -g "!package-lock.
json" -g "!.environ" -g "!*pot"'

**To Reproduce**

1. Create the minimal vimrc `min.vim`:

```vim
set nocompatible
set runtimepath^=/path/to/vim-clap
syntax on
filetype plugin indent on
```

2. Start (neo)vim with command: `nvim`

3. Create test.txt file with a content below
```
What is Lorem Ipsum?
It is Lorem a long established fact fact fact fact fact fact fact Lorem Ipsum

Why do we use it?
It is a long established fact that a reader will be distracted Lorem Ipsum
```

4. Invoke Clap grep and search "Lorem"

First line of results contains `What is Lorem Ipsum?` and preview is available and hitting enter leads to right placement in the file.

![searchLorem](https://user-images.githubusercontent.com/28489428/120536364-b61d3e00-c3ec-11eb-84e1-ec4566f73c0d.png)

Second line of results contains `It is Lorem a long established fact fact fact fact fact fact fact Lorem Ipsum` there is no preview!! and if you hit enter it will create new empty buffer rather than go to right placement.

![newEmptyBuffer](https://user-images.githubusercontent.com/28489428/120536657-072d3200-c3ed-11eb-8bae-6b9161bb69d4.png)

Third line - same faulty behavior as with the second line.
Both Second line and Third line contains 77 symbols, after deleting one symbol for line faulty behavior is gone for according line - you can preview and go to line from results. 

**Expected behavior**
Clap grep should work with a search result that is placed on a line with more than 77 symbols.

**Comment by romgrk** [comment]
It's creating a new buffer because it's taking the icon as part of the filename.

**Comment by liuchengxu** [comment]
Can not reproduce with the minimal vimrc. Please provide the true minimal vimrc to help reproduce.

**Comment by LoydAndrew** [comment]
You can reproduce with this minimal config.
There is some problem around relations with devicons - without it, it works.

```vim
set nocompatible
set showmatch
set ignorecase
set mouse=n
set hlsearch
set tabstop=4
set softtabstop=4
set expandtab
set shiftwidth=4
set autoindent
set smartindent
set number
set cc=80
set pyxversion=3
set path=$PWD/**
set t_Co=256
set cursorline
set termguicolors
syntax on

set relativenumber
set list
set wildoptions+=pum
set conceallevel=0
set wildignore+=*.o,*.obj,.git,*.rbc,*.pyc,*__pycache__,

if !filereadable('/tmp/plug.vim')
  silent !curl --insecure -fLo /tmp/plug.vim
    \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
endif

source /tmp/plug.vim
call plug#begin('/tmp/plugged')
Plug 'liuchengxu/vim-clap', { 'do': ':Clap install-binary'  }
Plug 'ryanoasis/vim-devicons'
call plug#end()

autocmd VimEnter * PlugClean! | PlugUpdate --sync | close
set background=dark " for the dark version
set termguicolors

let g:vim_json_conceal=0
let g:vim_json_syntax_conceal = 0
let g:vim_markdown_conceal = 0

let g:clap_theme = {
            \ 'display': {'ctermbg': '239', 'guibg': '#2d3135', 'guifg': '#f8f8f2', 'ctermfg': '123' },
            \ 'preview': {'ctermbg': '239', 'guibg': '#2d3135', 'guifg': '#f8f8f2', 'ctermfg': '123' }
            \}
"
" color= #282C34"
" guifg=#ABB2BF guibg=#282C34

let g:clap_preview_size = 10
let g:clap_enable_background_shadow = 0
let g:clap_insert_mode_only = 1
let g:clap_layout = { 'relative': 'editor' }
let g:clap_provider_grep_opts = '--vimgrep --hidden -g "!node_modules/" -g "!idea" -g "!.vscode" -g "!venv" -g "!.mypy_cache" -g "!build" -g "!dist"  -g "!__pycache__" -g "!tags" -g "!*.git" -g "!package-lock.json" -g "!.environ" -g "!*pot"'

let g:clap_enable_debug = 1
```

**Comment by liuchengxu** [comment]
@LoydAndrew Should be fixed by #702.
